wagtail/api/v2/views.py

```diff
@@ -453,6 +455,14 @@ def find_object(self, queryset, request):
         site = Site.find_for_request(request)
         if 'html_path' in request.GET and site is not None:
             path = request.GET['html_path']
+
+            # Check whether there is a matching page redirect and set path accordingly.
+            try:
```

Comment:
This check should only happen if the page routing has failed. So if there is both a page and a redirect for the same URL, we get the page.

Reply:
Hi Karl! Thank you for looking into this. I have made some changes to reflect your proposals. I placed a TODO where I thought would be a good place to add the header, because I did not know how to set it.